src/models/User.ts

```diff
@@ -1,5 +1,25 @@
 export interface User {
-    name: string,
-    email: string,
+    name: string
+    email: string
+    admin: boolean
     color?: string
-}
\ No newline at end of file
+}
+
+export function mapAuthResponseToUser(data: any): User {
+    return {
+        name: data.displayName || "",
+        email: data.email || "",
+        color: data.color || "",
+        admin: data.admin || false
+    }
+}
+
+export function mapResponseToUser(data: any): User {
```

Comment:
Hey, how about renaming this to mapResponseDbToUser. At first it took me a while to realize this is used to map the user from the db to our user model